misc/package/Data Files/MWSE/core/lib/logger/init.lua

```diff
@@ -0,0 +1,658 @@
+1) `log = Logger("MODNAME")`
+2) `log = Logger{modName="MODNAME", level=LEVEL?, ...}`
+3) `log = Logger.new("MODNAME")`
+4) `log = Logger.new{modName="MODNAME", level=LEVEL?, ...}`
+
+tring representing a numerical code (ie "NONE", "PROBLEMS", ..., "TRACE")
+
+### Writing log messages
+
+- you can log things by writing `log:info`, `log:warn`, `log:error`, `log:debug`, etc.
+- additionally, you can write debug messages by typing `log(str)`
+
+**Passing multiple parameters:**
+If multiple parameters are passed, then `string.format` will be called on the first parameter. You have two options for formatting strings:
+1) pass the formatting options as regular arguments.
+    - e.g., `log:trace("The %s today is %i %s", "weather", 20, "degrees")`
+    - So, `log:trace(s, ...)` and `log:trace(string.format(s, ...))` will print the same things.
+    - the only difference is that in the first case, `string.format` will be called AFTER the `log.level` is checked.
+2) pass the formatting options as a `function`.
+    - e.g., `log:trace("The %s today is %i, %s", function() return "weather", 20, "degrees" end)
+    - So, `log:trace(s, func)` and `log:trace(string.format(s, func() )) will print the same things.
+    - The key difference is that both `string.format` AND `func` will only be called after the `log.level` is checked.
+    - This could be very nice for performance reasons, if you're printing complicated debugging messages.
+
+
+**Passing a single parameter:**
+If only a single parameter is passed, `string.format` will NOT be called. This is to avoid unexpected errors from writing strings 
+that inadvertently contain formatting specifiers.
+
+
+
+### Updating the `log.level`
+
+Once a `log` has been created, the log level can be changed by using the `log:setLevel(Logger.LEVEL)` method.
+You can specify a string or number.
+e.g. to set the `log.level` to "DEBUG", you can write any of the following:
+1) `log:setLevel("DEBUG")`
+2) `log:setLevel(2)`
+3) `log:setLevel(Logger.LEVEL.DEBUG)`
+
+### Accessing the `log.level`
+
+The current logging level can be accessed by writing `log.level` or `#log`.
+
+The `log.level` is stored internally as an `integer`, so that you can easily check to see if the `log.level` is above a certain value.
+The log levels are:
+NONE = 0
+ERROR = 1
+WARN = 2
+INFO = 3
+DEBUG = 4
+TRACE = 5
+
+So, to check if the logging level is "DEBUG" or higher, you can write
+- `#log >= 4`
+
+Additionally, you can write
+- `log >= 4`
+
+**NOTE:** The `log` **must** be on the `>=` side of the comparison if using this syntax. Writing `log <= 4` will result in an error. 
+]]
+---@class mwseLogger
+---@field modName string the name of the mod this logger is for
+---@field level mwseLogger.LEVEL
+---@field moduleName string? the module this logger belongs to, or `nil` if it's a general purpose logger
+---@field useColors boolean should colors be used when writing log statements? Default: `false`
```

Comment:
Can you drop this field in favor of doing `if mwse.getConfig("EnableLogColors") then` directly in code to stay the same as it works now?

Reply:
i made this change, as well as the other changes you proposed (except the `parentLoggers` thing since i wasn't sure how to move forward on that one)